src/prefect/_internal/compatibility/async_dispatch.py

```diff
@@ -63,10 +63,10 @@ def _is_acceptable_callable(
 
 def async_dispatch(
     async_impl: Union[
-        Callable[P, Coroutine[Any, Any, R]],
-        "classmethod[type[Any], P, Coroutine[Any, Any, R]]",
+        Callable[..., Coroutine[Any, Any, R]],
+        "classmethod[type[Any], ..., Coroutine[Any, Any, R]]",
```

Comment:
this is because the sync and async versions of `run_deployment` need to take different types of clients. i could make them both `Optional[Any]` but thought that would be more unclear and this way it pushes folks who care about typing to use the explicit async one